### PR TITLE
Fixes #566 - adds some additional docs around additionalHeadersRef

### LINKED ISSUE #566: Additional Probe Headers could do with some docs
Follow up from 

https://github.com/Kuadrant/multicluster-gateway-controller/issues/501
and
https://github.com/Kuadrant/multicluster-gateway-controller/issues/515

The docs aren't particularly clear about how to use `additionalHeadersRef`.

### PR DESCRIPTION
**Pull request by jasonmadigan**
Fixes #566 - adds additional docs for additionalHeadersRef

### CODE REVIEW

**Comment by jasonmadigan** [comment]
@philbrookes would appreciate your thoughts on this one - added some extra docs on `additionalHeadersRef`

**Comment by jasonmadigan** [comment]
@david-martin may also be of interest

File: docs/dnspolicy/dns-health-checks.md [hunk 1]
```diff
@@ -45,6 +45,43 @@ EOF
 ```
 This configuration sets up a DNS health check by creating DNSHealthCheckProbes for the specified `prod-web` Gateway endpoints.
 
+### `additionalHeadersRef`
+
+The `additionalHeadersRef` field specifies a `Secret` used for storing supplementary HTTP headers. These headers are included when sending probe requests and can contain critical information like authentication tokens.
```
Comment (philbrookes) [inline docs/dnspolicy/dns-health-checks.md:50]: might be worth mentioning that the secret must be in the same namespace as the DNS Policy

**Comment by philbrookes** [comment]
looks good other than one small comment.

File: docs/dnspolicy/dns-health-checks.md (hunk 1 above)
Comment (jasonmadigan) [reply docs/dnspolicy/dns-health-checks.md:50]: good point, will mention

**Review by maleck13 — APPROVED** [review]

**Pull request merged**